[Mounil Sankar] ip

mounilsankar opened the pull request:
### NINO

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Nino frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/mounilsankar/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `greet` method:

```Java
    public static void greet() {
        String logo = "       .__\n"
                + "  ____ |__| ____   ____\n"
                + "/    \\|  |/    \\ /  _  \\\n"
                + "|   |  \\  |   |  (  <_> )\n"
                + "|___|  /__|___|  /\\____/\n"
                + "     \\/        \\/";
        System.out.println("Hello from\n" + logo);
        Ui.showLine();
        System.out.println("Hello! I'm NINO!");
        System.out.println("What can I do for you?");
        Ui.showLine();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

mounilsankar committed:
Level-0

mounilsankar committed:
Level-1

mounilsankar committed:
Level-0

mounilsankar committed:
Merge remote-tracking branch 'origin/master'

mounilsankar committed:
Level-2

mounilsankar committed:
Level-3

mounilsankar committed:
Level-4

mounilsankar committed:
A-TextUiTesting

mounilsankar committed:
Level-5

mounilsankar committed:
Level-6

mounilsankar committed:
Implemented Level-7

mounilsankar committed:
Merge branck-Level-7 to master

mounilsankar committed:
Merge branch 'branch-Level-7' with master

mounilsankar committed:
Implemented Level-8

mounilsankar committed:
Merge branch 'branch-Level-8' with master

mounilsankar committed:
Implemented A-MoreOOP

mounilsankar committed:
Merge branch 'branch-A-moreOOP' with master

mounilsankar committed:
Implemented A-Packages

mounilsankar committed:
Merge branch 'branch-A-Packages' with master

mounilsankar committed:
Merge branch 'add-gradle-support' to master

mounilsankar committed:
Implemented A-Gradle

mounilsankar committed:
Implemented A-JUnit

mounilsankar committed:
Implemented A-JavaDoc

mounilsankar committed:
Merge branch 'branch-A-JavaDoc' with master

mounilsankar committed:
Implemented A-CodingStandard

mounilsankar committed:
Implemented Level-9

mounilsankar committed:
Merge branch 'branch-A-CodingStandard' to master

mounilsankar committed:
Merge branch 'branch-Level-9' with master

mounilsankar committed:
Implemented A-Checkstyle

DonovanJJ commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+        }
+    }
+    private void run() {
+        Ui.greet();
```
Is there any reason why this is a class method instead of instance method

DonovanJJ commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    /**
+     * Main method for the Duke instance.
```
Good description regarding the method purpose and parameter type.
However, I think a new line at this portion of the javadoc would make your documentation seem less cluttered.

DonovanJJ commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,52 @@
+    public static Command parse(String fullCommand) {
+        //read first word in command
+        String[] words = fullCommand.split(" ", 2);
+        switch (words[0]) {
```
Great job on following the coding standard here 👍

DonovanJJ commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,102 @@
+            int taskNumber = i + 1;
+            Task t = this.taskList.get(i);
+            System.out.println(taskNumber
+                    + "."
```
Great that you broke this into multiple lines with proper coding standard! Helps a lot with the readability 👍

DonovanJJ commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,66 @@
+     * @param fullCommand The full command string as entered by the user.
+     */
+    public DeadlineCommand(String fullCommand) {
+
```
Should this new line be removed?

DonovanJJ commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,66 @@
+                }
+            }
+        }
+    }
```
Perhaps a new line here would improve code readability?
Seems to be the case throughout the Command subclasses

DonovanJJ commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,66 @@
+    /**
+     * Executes the deadline command, adding a new deadline task to the task list.
+     *
+     * @param tasks   The task list to which the deadline task will be added.
```
There seems to be a bit of inconsistency in spacing.

DonovanJJ reviewed (commented):
Overall very good work 👍 
Had an easy time reading and understanding your code, just some very minor lack of spaces here and there

mounilsankar committed:
Implemented Level-10

mounilsankar committed:
Added colour to dialog boxes

mounilsankar committed:
Added colour to dialog boxes

mounilsankar committed:
Merge branch 'branch-Level-10' with master

tayian commented:
interesting choice of name for your project (hope it means something meaningful for you). thumbs up for your implementation of OOP, and consistent use of JavaDoc to provide code clarity. May this tool ends up being useful for you as well!

don't forget to tick the features in your pull request description once they have been implemented!

tayian commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,95 @@
+        case("E"):
+            return Event.readFromFile(components);
+        default:
+            return null;
```
Amazing use of switch cases in the lines preceding this. Keep up the good work!

tayian commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,39 @@
+        tasks.printTasks();
+        Ui.showLine();
+    }
+    @Override
```
I noticed you have a 

```
    @Override
    public boolean isExit() {
        return false;
    }
```

for almost every exception. Perhaps there is a way to communicate that _this command is not the exit command_ without these lines of code that will have to be copy-pasted into each new command?

tayian commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,39 @@
+    @Override
+    public boolean isExit() {
+        return false;
+    }
```
I noticed you have a 

```
    @Override
    public boolean isExit() {
        return false;
    }
```

for almost every exception. Perhaps there is a way to communicate that _this command is not the exit command_ without these lines of code that will have to be copy-pasted into each new command?

mounilsankar committed:
Modified build gradle for new JAR file creation

mounilsankar committed:
Implemented A-Assertions

mounilsankar committed:
Implemented A-CodeQuality

mounilsankar committed:
Merge pull request #1 from mounilsankar/branch-A-Assertions

Add Java Assertions

mounilsankar committed:
Merge pull request #2 from mounilsankar/branch-A-CodeQuality

Refactored code to improve readability

mounilsankar committed:
Implement A-CI

 Introduced the CI workflow for Java.

 This is necessary to automate the build and check process for the project, to ensure consistent and efficient testing.

The workflow code has been added to the repository to enable automated building and checking of the project using Gradle on multiple platforms.

 This approach was chosen to streamline the CI process and improve development efficiency by automating the build and check steps.

 The new CI workflow is expected to enhance the development workflow by automating tasks that were previously performed manually, saving time and reducing the likelihood of errors.

mounilsankar committed:
Merge pull request #3 from mounilsankar/branch-A-CodeQuality

Implement A-CI

mounilsankar committed:
Implement A-DetectDuplicates

Add feature to detect duplicate tasks when adding to the task list and display error message if user does so

It's important to prevent duplicate tasks in the todo list as they can lead to confusion and inefficiency in task management.

This commit introduces a new feature that checks for duplicates when adding tasks to the todo list. If a duplicate task is detected, the user will be alerted through an error message and the task will not be added.

This approach was chosen to enhance the usability of the todo list application by providing a user-friendly way to avoid duplicate tasks. It improves the overall user experience and helps maintain a more organized todo list.

{any other relevant info} - This feature contributes to better task management and ensures that the todo list remains clutter-free and efficient. Users will now have a smoother and more effective experience when using the application.

mounilsankar committed:
Revert "Implement A-DetectDuplicates"

This reverts commit 1293faaf09cd43e7f54d9b2fa4cd12724ba628ae.

mounilsankar committed:
Implement C-DetectDuplicates

Add feature to detect duplicate tasks when adding to the task list and display error message if user does so

It's important to prevent duplicate tasks in the todo list as they can lead to confusion and inefficiency in task management.

This commit introduces a new feature that checks for duplicates when adding tasks to the todo list. If a duplicate task is detected, the user will be alerted through an error message and the task will not be added.

This approach was chosen to enhance the usability of the todo list application by providing a user-friendly way to avoid duplicate tasks. It improves the overall user experience and helps maintain a more organized todo list.

{any other relevant info} - This feature contributes to better task management and ensures that the todo list remains clutter-free and efficient. Users will now have a smoother and more effective experience when using the application.

mounilsankar committed:
Implement bugfix for Level-7 save feature

Initially the tasks were only being saved after each bye command

mounilsankar committed:
Implemented minor code quality improvements

mounilsankar committed:
Implemented minor code quality improvements

mounilsankar committed:
Implement A-UserGuide

mounilsankar committed:
Update doce/README.md

mounilsankar committed:
Storage bugfix

mounilsankar committed:
Update build.gradle
